src-v2/NTS.Persistance/Adapters/SetupLapRepository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using NTS.Domain.Setup.Entities;
+using NTS.Persistence.Setup;
+
+namespace NTS.Persistence.Adapters;
+
+public class SetupLapRepository : SetRepository<Lap, SetupState>
+{
+    public SetupLapRepository(IStore<SetupState> store) : base(store)
+    {
+        state = store.Load().Result;
+    }
+
+    private SetupState state { get; set; }
+    public override Task<Lap> Update(Lap entity)
+    {
+        state.Event.Competitions
+            .Where(competition => competition.Phases.Any(phase => phase.SelectedLap.Id == entity.Id))
+            .SelectMany(competition => competition.Phases
+                .Where(phase => phase.SelectedLap.Id == entity.Id)
+                .Select(phase => new { competition, phase }))
+            .ToList()
+            .ForEach(item =>
+            {
+                item.phase.SetSelectedLap(entity);
+                item.competition.Update(item.phase);
+            });
```

Comment:
Isn't
```c#
foreach (var phase in Event.Competitions.SelectMany(x => x.Phases))
{
    if (phase == entity)
    {
        phase.SetSelectedLap(entity);
    }
}
```
Simpler?

Additionally, the first `Where` clause is redundant as you are making the same check again in `SelectMany`.  Also `ToList` isn't necessary.